### PR TITLE
Rf-280 Only render avatar when audio visualizer gain difference is over a threshold

### LINKED ISSUE #280: [Refactor]: Only render avatar when audio visualizer gain difference is over a threshold
### Refactor Request

Currently, the avatar is being re-rendered every frame. I believe this is unnecessary.

### Describe the proposed solution

onAudioActivity should only be triggered when the gain difference is significantly larger than the last frame.
Jitters allow users to know their mic is working, but I don't think tiny jitters are necessary.

### PR DESCRIPTION
**Pull request by siberowl**
## Which Issue was this PR made for?

- Closes #280 

## What is within the scope with this PR

Refactor to lower the frequency of onAudioActivity getting called and improving performance.

### CODE REVIEW

**Comment by siberowl** [comment]
Hmmm it's definitely more performant but I feel like it's less responsive and less satisfying

**Comment by siberowl** [comment]
> Hmmm it's definitely more performant but I feel like it's less responsive and less satisfying

Realized it was because I wasn't updating the previous average at the correct location.

File: client/src/classes/AudioVisualizer.ts [hunk 1]
```diff
@@ -33,13 +37,16 @@ export class AudioVisualizer {
...
-  const max = 150;
-  let _multiplier = 0;
-  if (gain < max) {
-    _multiplier = gain / max;
-  } else {
-    _multiplier = 1;
-  }
-  return _multiplier;
+  const max = 250;
+  return gain < max ? gain / max : 1;
```
Comment (Njima1572) [inline client/src/classes/AudioVisualizer.ts:51]: It's more intuitive using Math.min(gain/max, 1). Just a personal preference.

Comment (siberowl) [reply client/src/classes/AudioVisualizer.ts:51]: I agree

**Pull request merged**